misc/testerina/modules/testerina-compiler-plugin/src/main/java/org/ballerinalang/testerina/compiler/TestExecutionGenerationTask.java

```diff
@@ -58,49 +50,28 @@
  */
 public class TestExecutionGenerationTask implements GeneratorTask<SourceGeneratorContext> {
 
-    private static final String TEST_REGISTER_FUNCTION = "registerTest";
-    private static final String SET_OPTIONS_FUNCTION = "setTestOptions";
-    private static final String START_SUITE_FUNCTION = "startSuite";
-    private static final String TEST_ORG_NAME = "ballerina";
-    private static final String TEST_MODULE_NAME = "test";
-    private static final String TEST_EXEC_FUNCTION = "__execute__";
-    private static final String TEST_EXEC_FILENAME = "test_execute";
-
-    private static final String TARGET_PATH_PARAMETER = "targetPath";
-    private static final String MODULE_NAME_PARAMETER = "moduleName";
-    private static final String REPORT_PATH_PARAMETER = "report";
-    private static final String COVERAGE_PATH_PARAMETER = "coverage";
-    private static final String GROUPS_PARAMETER = "groups";
-    private static final String DISABLE_GROUPS_PARAMETER = "disableGroups";
-    private static final String TESTS_PARAMETER = "tests";
-    private static final String RERUN_FAILED_PARAMETER = "rerunFailed";
-    private static final String LIST_GROUPS_PARAMETER = "listGroups";
-
     @Override
     public void generate(SourceGeneratorContext generatorContext) {
+        if (TesterinaCompilerPluginUtils.isSingleFileProject(generatorContext.currentPackage().project())) {
```

Comment:
This will be skipped by the compiler plugin manager. We dont have to handle it here